components/gitpod-protocol/src/util/context-url.ts

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2020 TypeFox GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+/**
+ * The field "contextUrl" might contain prefixes like:
+ *  - envvar1=value1/...
+ *  - prebuild/...
+ * This is the analogon to the (Prefix)ContextParser structure in "server".
+ */
+export function contextUrlToUrl(contextUrl: string): URL {
```

Comment:
given the call sites it should be:
`contextUrlToUrl(contextUrl: string | undefined): URL | undefined`

Reply:
Absolutely correct. Thank you!